Sources/BSON/BSONDocumentIterator.swift

```diff
@@ -108,6 +140,52 @@ public struct BSONDocumentIterator: IteratorProtocol {
             }
         }
     }
+
+    // uses an iterator to copy (key, value) pairs of the provided document from range [startIndex, endIndex) into a new
+    // document. starts at the startIndex-th pair and ends at the end of the document or the (endIndex-1)th index,
+    // whichever comes first.
+    internal static func subsequence(
```

Comment:
This static method looks like it would be better as a regular method, though it seems to have been written like this in the original BSON library for a while. Do you recall the reasoning for that @kmahar ?